### PR TITLE
Fix/6908 prevent loading UI on bg task

### PR DESCRIPTION
**Pull request by ArturFriesen**
## Description
The solution for the bug and the main change of this PR is to postpone `setupUI(...)` from `didFinishLaunchingWithOptions` to `didBecomeActive`.

`didFinishLaunchingWithOptions` is called when the app state changes from `not running` / `suspended` to `background`.
`setupUI(...)` leads to some `viewDidLoad` calls, which lead to updating the test results (and also other calls).

In case of a background task execution there was a race condition between the update starting from `viewDidLoad` and the update starting from the background task. Due to this race condition the alert was not triggered sometimes.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-6908

### CODE REVIEW

File: src/xcode/ENA/ENA/Resources/Info_Debug.plist [hunk 1]
```diff
@@ -56,6 +56,8 @@
 	<true/>
 	<key>UILaunchStoryboardName</key>
 	<string>LaunchScreen</string>
+	<key>UIMainStoryboardFile</key>
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Resources/Info_Debug.plist:59]: Do we need this change in the other Info_*.plist as well?

**Review by marcussc — APPROVED** [review]
Nice finding!

File: src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate.swift [hunk 2]
```diff
@@ -118,8 +120,9 @@ class AppDelegate: UIResponder, UIApplicationDelegate, CoronaWarnAppDelegate, Re
 		}
 
 		// Check for any URLs passed into the app – most likely via scanning a QR code from event or antigen rapid test
-		let route = routeFromLaunchOptions(launchOptions)
-		setupUI(route)
+		// Route will be executed in 'applicationDidBecomeActive'
+		self.route = routeFromLaunchOptions(launchOptions)
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate.swift:124]:
```suggestion
		route = routeFromLaunchOptions(launchOptions)
```

File: src/xcode/ENA/ENA/Resources/Info_Debug.plist (hunk 1 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Resources/Info_Debug.plist:59]: Good point!

File: src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate.swift [hunk 3]
```diff
@@ -156,7 +160,12 @@ class AppDelegate: UIResponder, UIApplicationDelegate, CoronaWarnAppDelegate, Re
 	}
 
 	func applicationDidBecomeActive(_ application: UIApplication) {
-		Log.info("Application did become active.", log: .background)
+		Log.info("Application did become active.", log: .appLifecycle)
+
+		if !didSetupUI {
+			setupUI(route)
+			didSetupUI = true
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate.swift:167]: i think you should also reset `route` after handling it ti avoid it gets handled twice

**Review by KaiTeuber — APPROVED** [review]
Did you check if it handles the `postOnboardingRoute` correctly? I guess it will, but I'm not sure about that.

File: src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate.swift (hunk 3 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/AppDelegate & Globals/AppDelegate.swift:167]: Good point. Done.

File: src/xcode/ENA/ENA/Resources/Info_Debug.plist (hunk 1 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Resources/Info_Debug.plist:59]: done

**Review by 30mar — APPROVED** [review]

**Pull request merged**